hooks/gen_docs.py

```diff
@@ -1,13 +1,336 @@
+
+
+@dataclass
+class EnvVar:
+    """Environment variable's properties."""
+
+    name: str
+    default_value: Any
+    required: bool
+    description: str | None
+    corresponding_setting_name: str | None
+
+
+class EnvVarAttrs(str, Enum):
+    """The attr names are used as columns for the markdown tables."""
+
+    NAME = "Name"
+    DEFAULT_VALUE = "Default Value"
+    REQUIRED = "Required"
+    DESCRIPTION = "Description"
+    CORRESPONDING_SETTING_NAME = "Setting name"
+
+
+def csv_append_env_var(
+    file: Path,
+    name: str,
+    default_value: Any,
+    description: str | list[str] | None,
+    *args: Any,
+) -> None:
+    """Append env variable record to a chosen .csv file, create it if doesn't exist.
+
+    :param file: csv_file
+    :param name: env var name
+    :param default_value: Default value of the env var
+    :param description: Text description of the env var
+    :param args: Any additional fields to be added to the csv file
+    """
+    formatted_description = ""
+    if description is not None:
+        if isinstance(description, str):
+            description = [description]
+        for section in description:
+            formatted_description += fill(
+                text=section,
+                width=68,
+            )
+    required = False
+    if default_value == Ellipsis:
+        required = True
+        default_value = ""
+    elif default_value is None:
+        default_value = ""
+    with file.open("a+") as csv_file:
+        csv.writer(csv_file).writerow(
+            [name, default_value, required, formatted_description, *list(args)],
+        )
+
+
+def write_title_to_dotenv_file(
+    file_path: Path,
+    title: str,
+    description: str,
+    comment_symbol: str = "#",
+) -> None:
+    """Overwrite a file with a title and description as comments.
+
+    All previous contents will be erased!
+
+    Example output:
+
+    .. code-block:: python
+
+        # Title
+        #
+        # Multi-line description of length 70
+        # second line
+        #
+        VAR = None
+
+    :param file_path: Path to the file
+    :param title: Title
+    :param description: File description
+    :param comment_symbol: Symbol to use for comments, defaults to "#"
+    """
+    with file_path.open("w+") as f:
+        text = (
+            f"{comment_symbol} {title}"
+            f"\n{comment_symbol}\n"
+            + fill(
+                text=description,
+                initial_indent=f"{comment_symbol} ",
+                subsequent_indent=f"{comment_symbol} ",
+            )
+            + f"\n{comment_symbol}\n"
+        )
+        f.write(text)
+
+
+def append_csv_to_dotenv_file(
+    source: Path,
+    target: Path,
+) -> None:
+    """Parse csv file looking for a env variable records and append them into a dotenv file.
+
+    :param source: csv file to read from
+    :param target: dotenv file to wread into
+    """
+    comment_symbol = "#"  # Comments in .env start with a `#`
```

Comment:
could be defined as a constant outside this function, also re-use it in `write_title_to_dotenv_file`